main.go

```diff
@@ -3,33 +3,57 @@ package main
 import (
 	"fmt"
 	"os"
-)
-
-var version string
-var Usage = fmt.Sprintf(`Usage: lemonade [options]... SUB_COMMAND [arg]
-Sub Commands:
-  open [URL]                  Open URL by browser
-  copy [text]                 Copy text.
-  paste                       Paste text.
-  server                      Start lemonade server.
-
-Options:
-  --port=2489                 TCP port number
-  --allow="0.0.0.0/0,::/0"    Allow IP Range             [Server only]
-  --host="localhost"          Destination hostname       [Client only]
-  --trans-loopback=true       Translate loopback address [open subcommand only]
-  --trans-localfile=true      Translate local file path  [open subcommand only]
-  --help                      Show this message
 
-
-Version:
-  %s`, version)
+	"github.com/pocke/lemonade/client"
+	"github.com/pocke/lemonade/lemon"
+	"github.com/pocke/lemonade/server"
+)
 
 func main() {
-	cli := &CLI{
+
+	cli := &lemon.CLI{
 		In:  os.Stdin,
 		Out: os.Stdout,
 		Err: os.Stderr,
 	}
-	os.Exit(cli.Do(os.Args))
+	os.Exit(Do(cli, os.Args))
+}
+
+func Do(c *lemon.CLI, args []string) int {
+	if err := c.FlagParse(args); err != nil {
+		writeError(c, err)
+		return lemon.FlagParseError
+	}
+	if c.Help {
+		fmt.Fprint(c.Err, lemon.Usage)
+		return lemon.Help
+	}
+
+	lc := client.New(c)
+	var err error
+
+	switch c.Type {
+	case lemon.OPEN:
+		err = lc.Open(c.DataSource, c.TransLocalfile, c.TransLoopback)
+	case lemon.COPY:
+		err = lc.Copy(c.DataSource)
+	case lemon.PASTE:
+		var text string
+		text, err = lc.Paste()
+		c.Out.Write([]byte(text))
```

Comment:
__[gometalinter]__
* [errcheck] error return value not checked (c.Out.Write([]byte(text)))